compiler/rustc_ast_lowering/src/expr.rs

```diff
@@ -655,33 +655,40 @@ impl<'hir> LoweringContext<'_, 'hir> {
 
             hir::ExprKind::Closure(c)
         };
-        let parent_has_track_caller = self
-            .attrs
-            .values()
-            .find(|attrs| attrs.into_iter().find(|attr| attr.has_name(sym::track_caller)).is_some())
-            .is_some();
+
         let unstable_span =
             self.mark_span_with_reason(DesugaringKind::Async, span, self.allow_gen_future.clone());
 
-        let hir_id = if parent_has_track_caller {
-            let generator_hir_id = self.lower_node_id(closure_node_id);
-            self.lower_attrs(
-                generator_hir_id,
-                &[Attribute {
-                    kind: AttrKind::Normal(ptr::P(NormalAttr {
-                        item: AttrItem {
-                            path: Path::from_ident(Ident::new(sym::track_caller, span)),
-                            args: AttrArgs::Empty,
+        let hir_id = if self.tcx.features().closure_track_caller {
+            let parent_has_track_caller = self
+                .attrs
+                .values()
+                .find(|attrs| {
+                    attrs.into_iter().find(|attr| attr.has_name(sym::track_caller)).is_some()
+                })
+                .is_some();
```

Comment:
nit: `find + is_some = any`

```suggestion
                .any(|attrs| {
                    attrs.iter().any(|attr| attr.has_name(sym::track_caller))
                });
```

Reply:
thanks :) much cleaner!